src/core/dev/Encoder.cpp

```diff
@@ -114,17 +114,23 @@ void Encoder::setRangeAndPosition(uint32_t newRange, uint32_t newPosition) {
 }
 
 int8_t Encoder::readPinValues() {
-    bool aPos = (bool) a.readPin();
-    bool bPos = (bool) b.readPin();
-    // calculating and returning the position
-    if (aPos == 0 && bPos == 0) {
+    uint8_t aPos = (bool) a.readPin();
+    uint8_t bPos = (bool) b.readPin();
+    // Calculating and returning the position.
+    // Position is in graycode, not binary.
+    uint8_t comb = (bPos << 1) | aPos;
+    switch (comb) {
+    case 0b00:
         return 0;
-    } else if (aPos == 1 && bPos == 0) {
+    case 0b01:
         return 1;
-    } else if (aPos == 1 && bPos == 1) {
+    case 0b11:
         return 2;
-    } else if (aPos == 0 && bPos == 1) {
+    case 0b10:
         return 3;
+    default:
+        // this default case will never really happen.
+        return 0;
```

Comment:
If this should never happen, make it return a value that is recognizably wrong, like -1

Reply:
I considered that, but the issue is if this fails, I think failing in a way that just doesn't register the encoder moving is far safer than failing in a way that causes the encoder to move super far makes more sense. Maybe I should add a log statement for the default case, but the issue is having it return -1 will not actually break anything, so it will still fail silently. Also it's literally impossible for the default case to happen. Like there are only 4 possible values that can even be calculated.

Reply:
The only way for it to get to the default case is if someone accidentally deletes one of the other cases, which is highly unlikely. However, I would argue that when someone does something really dumb, we should be loud about it.

For example if someone deletes the case for `0b10` and the default case returns 0, someone might think the encoder is just skipping position 3 for some reason and spend time debugging that. If instead it returned -1, it would be immediately obvious that something major is wrong with the driver

Reply:
Removed the default case and added a return 0; line outside of the switch statement to prevent the compiler complaining. Once again this return 0; outside of the switch will never happen.